homeassistant/components/rachio/coordinator.py

```diff
@@ -54,3 +66,55 @@ async def _async_update_data(self) -> dict[str, Any]:
         except Timeout as err:
             raise UpdateFailed(f"Could not connect to the Rachio API: {err}") from err
         return {valve[KEY_ID]: valve for valve in data[1][KEY_VALVES]}
+
+
+class RachioScheduleUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Coordinator for fetching hose timer schedules."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        rachio: Rachio,
+        base_station,
+    ) -> None:
+        """Initialize a Rachio schedule coordinator."""
+        self.hass = hass
+        self.rachio = rachio
+        self.base_station = base_station
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN} schedule update coordinator",
+            update_interval=timedelta(minutes=30),
+        )
+
+    async def _async_update_data(self) -> list[dict[str, Any]]:
+        """Retrieve data for the past week and the next 60 days."""
+        _time_start: datetime = dt_util.now() - timedelta(days=7)
+        _time_end: datetime = dt_util.now() + timedelta(days=60)
+        start: dict[str, int] = {
+            YEAR: _time_start.year,
+            MONTH: _time_start.month,
+            DAY: _time_start.day,
+        }
+        end: dict[str, int] = {
+            YEAR: _time_end.year,
+            MONTH: _time_end.month,
+            DAY: _time_end.day,
+        }
+
+        try:
+            schedule = await self.hass.async_add_executor_job(
+                self.rachio.summary.get_valve_day_views,
+                self.base_station[KEY_ID],
+                start,
+                end,
+            )
+        except Timeout as err:
+            raise UpdateFailed(f"Could not connect to the Rachio API: {err}") from err
+        events = []
+        # Flatten and sort dates
+        for event in schedule[1][KEY_DAY_VIEWS]:
+            events.extend(event[KEY_PROGRAM_RUN_SUMMARIES])
+        events.sort(key=lambda event: event[KEY_START_TIME])
```

Comment:
You can use `itemgetter` here instead of a lambda https://docs.python.org/3/library/operator.html#operator.itemgetter